src/histbuf.rs

```diff
@@ -146,32 +153,36 @@ where
     ///
     /// ```
     /// use heapless::HistoryBuffer;
-    /// use heapless::consts::*;
     ///
-    /// let mut x: HistoryBuffer<u8, U16> = HistoryBuffer::new();
+    /// let mut x: HistoryBuffer<u8, 16> = HistoryBuffer::new();
     /// x.write(4);
     /// x.write(10);
-    /// assert_eq!(x.recent(), &10);
+    /// assert_eq!(x.recent(), Some(&10));
     /// ```
-    pub fn recent(&self) -> &T {
+    pub fn recent(&self) -> Option<&T> {
         if self.write_at == 0 {
-            &self.data[self.len() - 1]
+            if self.filled {
+                Some(unsafe { &*self.data[self.capacity() - 1].as_ptr() })
+            } else {
+                None
+            }
         } else {
-            &self.data[self.write_at - 1]
+            Some(unsafe { &*self.data[self.write_at - 1].as_ptr() })
         }
     }
 
     /// Returns the array slice backing the buffer, without keeping track
     /// of the write position. Therefore, the element order is unspecified.
     pub fn as_slice(&self) -> &[T] {
-        &self.data
+        if self.filled {
+            unsafe { slice::from_raw_parts(self.data.as_ptr() as *const _, self.capacity()) }
```

Comment:
Oh no! UB detected (out of bounds).

It should be `unsafe { slice::from_raw_parts(self.data.as_ptr() as *const _, self.len()) }`. And no `if self.filled` is required.

Reply:
```suggestion
            unsafe { slice::from_raw_parts(self.data.as_ptr() as *const _, self.len()) }
```